extensions/amp-form/0.1/amp-form.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+import {user} from '../../../src/log';
+import {xhrFor} from '../../../src/xhr';
+import {childElements, childElement, removeChildren} from '../../../src/dom';
+import {templatesFor} from '../../../src/template';
+import {CSS} from '../../../build/amp-form-0.1.css';
+
+class AmpForm extends AMP.BaseElement {
+
+  /** @override */
+  preconnectCallback(onLayout) {
+    this.preconnect.url(this.action_, onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == 'container';
+  }
+
+  /** @override */
+  buildCallback() {
+    /** @const @private {!Templates} */
+    this.templates_ = templatesFor(this.getWin());
+
+    /** @const @private {!Element} */
+    this.container_ = this.getWin().document.createElement('div');
+    this.element.appendChild(this.container_);
+
+    /** @const @private {!Element} */
+    this.messagesContainer_ = this.getWin().document.createElement('div');
+    this.messagesContainer_.classList.add('amp-hidden');
+    this.container_.appendChild(this.messagesContainer_);
+
+    //this.messagesContainer_.appendChild();
+
+    /** @const @private {!Xhr} */
+    this.xhr_ = xhrFor(this.getWin());
+
+    /** @const @private {!Element} */
+    this.form_ = this.getWin().document.createElement('form');
+
+    /** @const @private {!Array<!Element>} */
+    this.fields_ = childElements(this.element, el => el.tagName == 'AMP-FIELD');
```

Comment:
This only moves the fields into a form wrapper. I am not entirely clear on the promise part, but if that's the case, what would be a good alternative about thinking about this? Do we change the format to be actually get the user to provide the `<form>` element as well?